llvm/lib/AsmParser/LLParser.cpp

```diff
@@ -3948,7 +3945,6 @@ bool LLParser::parseValID(ValID &ID, PerFunctionState *PFS, Type *ExpectedTy) {
     unsigned Flags = 0;
     if (NUW)   Flags |= OverflowingBinaryOperator::NoUnsignedWrap;
     if (NSW)   Flags |= OverflowingBinaryOperator::NoSignedWrap;
-    if (Exact) Flags |= PossiblyExactOperator::IsExact;
```

Comment:
Unrelated musing: I've been wondering for a long time if we couldn't turn `or` and `and` into possibly exact operators with the meaning of `exact` being "the set / unset bits of the operands are disjoint". The idea being that this might help sometimes with issues caused by LLVM eagerly replacing `add` with `or` in address expressions.

Reply:
+1 I've wondered something similar.

Reply:
Sounds like a good addition to me. At least for `or`, the utility is pretty clear. For `and`, I'm not sure where we could make use of this property.

Reply:
Yeah, I don't have a concrete use for `and`. The motivation is simply to close over deMorgan's laws.

Reply:
I'd like to work on a patch for this. Do we want to use 'exact' or have a new keyword like 'disjoint'?

Reply:
I would lean towards a new flag. The usage will be quite different from the existing `exact` flag.